diff --git a/drivers/soc/qcom/icnss2/main.c b/drivers/soc/qcom/icnss2/main.c
index 6693f65a3..9404f31a3 100644
--- a/drivers/soc/qcom/icnss2/main.c
+++ b/drivers/soc/qcom/icnss2/main.c
@@ -2,7 +2,7 @@
 /*
  * Copyright (c) 2015-2021, The Linux Foundation. All rights reserved.
  */
-
+#pragma GCC diagnostic ignored "-Wstrict-prototypes"
 #define pr_fmt(fmt) "icnss2: " fmt
 
 #include <linux/of_address.h>
